File: src/main/java/com/mattchowning/file_read_write/server/handler/FileManager.java
package com.mattchowning.file_read_write.server.handler;

import java.io.*;

import io.netty.buffer.ByteBuf;

public class FileManager {

    private final String fullFilePath;

    public FileManager(String fullFilePath) {
        this.fullFilePath = fullFilePath;
    }

    public File getFile() {
        return new File(fullFilePath);
    }

    public boolean writeFileContent(ByteBuf byteBuf) {
        try {
            byte[] bytes = getBytes(byteBuf);
            FileOutputStream fileStream = new FileOutputStream(getFile());
            fileStream.write(bytes);
            return true;
        } catch (IOException e) {
            e.printStackTrace();
            return false;
        }
    }

    private static byte[] getBytes(ByteBuf buf) {
        byte[] result = new byte[buf.readableBytes()];
        buf.getBytes(buf.readerIndex(), result);
        return result;
    }
}

File: src/main/scala/com/mattchowning_scala/file_read_write/server/handler/FileManager.scala
package com.mattchowning_scala.file_read_write.server.handler

import java.io._
import io.netty.buffer.ByteBuf

class FileManager(private val fullFilePath: String) {

  def getFile: File = {
    new File(fullFilePath)
  }

  def writeFileContent(byteBuf: ByteBuf): Boolean = {
    try {
      val bytes: Array[Byte] = getBytes(byteBuf)
      val fileStream: FileOutputStream = new FileOutputStream(getFile)
      fileStream.write(bytes)
      true
    } catch {
      case e: IOException =>
        e.printStackTrace()
        false
    }
  }

  private def getBytes(buf: ByteBuf): Array[Byte] = {
    val result: Array[Byte] = new Array[Byte](buf.readableBytes)
    buf.getBytes(buf.readerIndex, result)
    result
  }
}